File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        left = max_freq = res = 0
        freq = {}
        for right in range(0, len(s)):
            freq[s[right]] = freq.get(s[right], 0) + 1
            max_freq = max(max_freq, freq[s[right]])
            if not right-left-max_freq+1 <= k:
                freq[s[left]] -= 1
                left += 1
            res = right-left+1
        return res

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.ts
function characterReplacement(s: string, k: number): number {
    const freq: Record<string, number> = {}
    let l = 0, max = 0, res = 0
    for (let r=0; r < s.length; r++) {
        freq[s[r]] = freq[s[r]] + 1 || 1
        max = Math.max(max, freq[s[r]])
        if (!(r-l-max+1 <= k)) {
            freq[s[l++]]--
        }
        res = r-l+1
    }
    return res
};